### ISSUE TITLE
Starting a frame with \verb

### ISSUE THREAD

**Issue by josephwright** [josephwright/beamer#93]
Hi,

When I start a frame with a `\verb` command, the output looks wrong:
````
\documentclass{beamer}
\begin{document}
\begin{frame}[fragile]
\verb|hello -> world| is broken
\end{frame}
\end{document}
````

The result is identical in Beamer 3.06 in TeX Live 2007 (Fedora 11) and Beamer 3.10 in MiKTeX 2.9 (WinXP). Instead of `hello -> world`, the output says `|hello-¿ world-`

The problem can be circumvented by adding extra whitespace in front of the \verb, but I believe the current behavior is a bug.

Thanks for your work on `Beamer`,

Arni

----------------------------------------
- Bitbucket: https://bitbucket.org/rivanvx/beamer/issue/93
- Originally reported by: Anonymous
- Originally created at: 2010-10-29

**Comment by josephwright** [comment]
_From Joseph Wright on 2010-10-30_

Okay, this happens because `beamer` writes the fragile frame input to an intermediate file.

TeX is inserting an additional space after the first token in the frame as it is being read as a control sequence. So you get
````
\verb |hello -> world| is broken
````
whereas with an additional blank line you get
````
\par \verb|hello -> world| is broken
````
I'll take a look and try to find why the first token is being read in this way.

**Comment by josephwright** [comment]
_From Joseph Wright on 2010-10-30_

Okay, here is the technical analysis of the issue. `beamer` is pretty flexible about reading the first line of the frame, and allows a lot of variation in what might be present. The way that this is done is to look ahead for a number of different things. This uses the `\futurelet` primitive, which will tokenize whatever it reads. Thus the very first token after
````
\begin{frame}[fragile]
````
will have been tokenized before any steps have been taken by `beamer`. For everything else in the frame, the input is read verbatim, avoiding any issues. Now, the question is 'can we find a good way to reverse this?'.

**Comment by josephwright** [comment]
_From Joseph Wright on 2010-10-30_

Right, I have a possible solution. This might be a bit risky, so I think it will need some serious testing. I've attached a diff which shows the approach: basically, you need to sanitize the first token in the input. That requires some adjustments to the reading mechanism, space-stripping from control sequence input and and adjustment to how titles are read.
````
diff -r f39336aafb3c base/beamerbaseframe.sty
--- a/base/beamerbaseframe.sty	Tue Oct 26 20:09:31 2010 +0200
+++ b/base/beamerbaseframe.sty	Sat Oct 30 15:03:00 2010 +0100
@@ -542,7 +542,16 @@
   \immediate\openout\beamer@verbatimfileout=\beamer@verbfilename%
   \expandafter\beamer@checktitleverbatim\@gobble}
 \def\beamer@checktitleverbatim{\@ifnextchar\bgroup{\beamer@verbatim@frametitle}{\beamer@verbatimreadframe}}
-\def\beamer@verbatim@frametitle#1{\@ifnextchar\bgroup{\beamer@verbatim@framesubtitle{#1}}{\beamer@verbatimreadframe\frametitle{#1}}}
+\def\beamer@verbatim@frametitle#1{%
+  \@ifnextchar\bgroup{\beamer@verbatim@framesubtitle{#1}}
+  {%
+    \@temptokena{#1}%
+    \edef\@tempa{\string\frametitle\string{\the\@temptokena\string}}%
+    \@temptokena\expandafter{\@tempa}%
+    \immediate\write\beamer@verbatimfileout{\the\@temptokena}%
+    \beamer@verbatimreadframe
+  }%
+}
 \def\beamer@verbatim@framesubtitle#1#2{\beamer@verbatimreadframe\frametitle{#1}\framesubtitle{#2}}
 % will return to following command with frame contents written to \beamer@verbfilename
 \def\beamer@afterexternalframe{%
diff -r f39336aafb3c base/beamerbaseverbatim.sty
--- a/base/beamerbaseverbatim.sty	Tue Oct 26 20:09:31 2010 +0200
+++ b/base/beamerbaseverbatim.sty	Sat Oct 30 15:03:00 2010 +0100
@@ -37,17 +37,23 @@
 }
 
 {\catcode`\^^M=12\endlinechar=-1%
-  \long\gdef\beamer@processframefirstline#1^^M{%
-    \def\beamer@test{#1}%
+  \long\gdef\beamer@processframefirstline#1#2^^M{%
+    \def\beamer@test{#1#2}%
     \ifx\beamer@test\beamer@stopframefirst%
       \let\next=\beamer@framewritten%
     \else
-      \@temptokena={#1}%
-      \immediate\write\beamer@verbatimfileout{\the\@temptokena}%
+      \def\beamer@test{#1}%
+      \@onelevel@sanitize\beamer@test
+      \edef\beamer@test{\beamer@test\space}%
+      \edef\beamer@test{\expandafter\zap@space\beamer@test\@empty}%
+      \@temptokena{#2}%
+      \immediate\write\beamer@verbatimfileout{\beamer@test\the\@temptokena}%
       \let\next=\beamer@processframeline%
     \fi%
     \next%
   }
+  \long\gdef\beamer@processframefirstline@aux#1{%
+  }%
   \long\gdef\beamer@processframeline#1^^M{%
     \def\beamer@test{#1}%
     \ifx\beamer@test\beamer@stopframe%
````

**Comment by josephwright** [comment]
Fixed by d06e1c1acda5fa9e02e86e642ed1bcf94359f37d
